salt/modules/file.py

```diff
@@ -1651,8 +1651,13 @@ def replace(path,
         with salt.utils.fopen(path,
                               mode='r',
                               buffering=bufsize) as r_file:
+            count_replaced = 0
             for line in r_file:
-                result, nrepl = re.subn(cpattern, repl, line, count)
+                if count == 0 or count_replaced != count:
```

Comment:
``count_replaced`` cannot become greater than ``count`` since value of ``nrepl`` is added to ``count_replaced`` inside the loop. Tested this to work for multiple occurrences of a pattern in the same line.